cedar-lean/Cedar/Thm/Partial/Evaluation/AndOr.lean

```diff
@@ -0,0 +1,81 @@
+/-
+ Copyright Cedar Contributors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+-/
+
+import Cedar.Partial.Evaluator
+import Cedar.Thm.Data.Control
+import Cedar.Thm.Partial.Evaluation.Basic
+
+namespace Cedar.Thm.Partial.Evaluation.AndOr
+
+open Cedar.Data
+open Cedar.Partial (Unknown)
+
+/- ## Lemmas shared by And.lean and Or.lean -/
```

Comment:
Why do you need a new file here? Why can't you just define this in `And.lean` or `Or.lean` and then import the lemma in the other file, like how you do for Set & Call?

Reply:
I guess I slightly prefer reusing existing files over making a new file. But I don't feel strongly. Just wanted to know if there was some reason I was missing 🙂

Reply:
My instinct is generally to prefer many smaller files over fewer long files, but I don't feel too strongly either

Reply:
I think I'll leave it this way for this PR, and in a followup we can standardize on either one or the other approach